src/DynamoCore/DSEngine/NodeToCode.cs

```diff
@@ -738,6 +738,9 @@ public string GetNextName()
                         var inputVar = node.GetAstIdentifierForOutputIndex(i).Value;
                         outputMap[inputVar] = string.Empty;
                     }
+
+                    var previewVar = node.AstIdentifierForPreview.Value;
+                    outputMap[previewVar] = string.Empty;
```

Comment:
Agreed, N2C should be a service (something like a chain of post-processing operations) that a bunch of nodes pass through and become CBNs. But before `GetIntermediateVariables`, I think the general constructs for variable preview need some unification (for example, `public virtual GetAstIdentifierForOutputIndex()` that derived classes can choose to optionally override, and also `public virtual AstIdentifierBase`. Not to forget `AstIdentifierForPreview`, too! Those are confusing at best and they are error-prone. There needs to be a clean-up task for this, if none is already in place.